mongodbatlas/resource_mongodbatlas_project_api_key.go

```diff
@@ -73,42 +73,41 @@ type APIProjectAssignmentKeyInput struct {
 	RoleNames []string `json:"roles,omitempty"`
 }
 
+const errorNoProjectAssignmentDefined = "could not obtain a project id as no assignments are defined"
+
 func resourceMongoDBAtlasProjectAPIKeyCreate(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
 	conn := meta.(*MongoDBClient).Atlas
-	projectID := d.Get("project_id").(string)
-	createRequest := new(matlas.APIKeyInput)
 
 	var apiKey *matlas.APIKey
 	var err error
 	var resp *matlas.Response
 
+	createRequest := new(matlas.APIKeyInput)
 	createRequest.Desc = d.Get("description").(string)
 	if projectAssignments, ok := d.GetOk("project_assignment"); ok {
 		projectAssignmentList := ExpandProjectAssignmentSet(projectAssignments.(*schema.Set))
-		for _, apiKeyList := range projectAssignmentList {
-			if apiKeyList.ProjectID == projectID {
-				createRequest.Roles = apiKeyList.RoleNames
-				apiKey, resp, err = conn.ProjectAPIKeys.Create(ctx, projectID, createRequest)
-				if err != nil {
-					if resp != nil && resp.StatusCode == http.StatusNotFound {
-						d.SetId("")
-						return nil
-					}
-				}
+
+		// creates api key using project id of first defined project assignment
+		firstAssignment := projectAssignmentList[0]
```

Comment:
should this new behaviour be documented? is it a breaking change? What happens with `project_id` when it is actually defined? from this code it seems like it won't be used aynmore

Reply:
So this is more of an implementation detail: one project id is used for the creation of the api key, and then remaining projects assignments are assigned to the new api key. The previous implementation was essentially the same, only instead of taking the first element for the project assignment list it would grab the project which was defined in root project_id for the creation of the api key. From a users standpoint it is the same.

> What happens with project_id when it is actually defined? from this code it seems like it won't be used anymore

This is correct, it is no longer used. From docs changes and deprecation warning users will be aware this attribute is no longer being used (https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1665/files#diff-31571e18c14ed5632e87c050342b195d9cbc5092b465b08abfb8b1dcf87d37bdL49)